common/djangoapps/credo_modules/models.py

```diff
@@ -30,10 +30,11 @@ def users_with_additional_profile(cls, course_id):
         return result
 
 
-def user_must_fill_additional_profile_fields(course, user):
+def user_must_fill_additional_profile_fields(course, user, block):
+    graded = block.graded if block else False
     course_key = course.id
     if course.credo_additional_profile_fields and user.email.endswith('@credomodules.com') \
```

Comment:
Please move `graded` param to be the first in the `if` statement. Because it will be checked firstly and if `graded == False` more complicated statements (like `CourseEnrollment.is_enrolled(user, course_key)` that makes query to DB) will not be executed